algosdk/testing/teal_blackbox.py

```diff
@@ -0,0 +1,1010 @@
+from base64 import b64decode
+import csv
+from dataclasses import dataclass
+from enum import Enum, auto
+import io
+from inspect import signature
+from tabulate import tabulate
+from typing import Any, Callable, Dict, Iterable, List, Optional, Tuple, Union
+
+from algosdk.v2client.algod import AlgodClient
+from algosdk.testing.dryrun import (
+    ZERO_ADDRESS,
+    assert_error,
+    assert_no_error,
+    Helper as DryRunHelper,
+)
+
+
+class ExecutionMode(Enum):
+    Signature = auto()
+    Application = auto()
+
+
+class DryRunProperty(Enum):
+    cost = auto()
+    lastLog = auto()
+    finalScratch = auto()
+    stackTop = auto()
+    maxStackHeight = auto()
+    status = auto()
+    rejected = auto()
+    passed = auto()
+    error = auto()
+    errorMessage = auto()
+    globalStateHas = auto()
+    localStateHas = auto()
+
+
+DRProp = DryRunProperty
+
+
+@dataclass
+class TealVal:
+    i: int = 0
+    b: str = ""
+    is_b: bool = None
+    hide_empty: bool = True
+
+    @classmethod
+    def from_stack(cls, d: dict) -> "TealVal":
+        return TealVal(d["uint"], d["bytes"], d["type"] == 1, hide_empty=False)
+
+    @classmethod
+    def from_scratch(cls, d: dict) -> "TealVal":
+        return TealVal(d["uint"], d["bytes"], len(d["bytes"]) > 0)
+
+    def is_empty(self) -> bool:
+        return not (self.i or self.b)
+
+    def __str__(self) -> str:
+        if self.hide_empty and self.is_empty():
+            return ""
+
+        assert (
+            self.is_b is not None
+        ), f"can't handle StackVariable with empty type"
+        return f"0x{b64decode(self.b).hex()}" if self.is_b else str(self.i)
+
+    def as_python_type(self) -> Union[int, str, None]:
+        if self.is_b is None:
+            return None
+        return str(self) if self.is_b else self.i
+
+
+@dataclass
+class BlackBoxResults:
```

Comment:
@tzaffi I've opened an optional PR for naming consistency via https://github.com/algorand/py-algorand-sdk/pull/304.   Can you have a look?

Let's resolve the thread once https://github.com/algorand/py-algorand-sdk/pull/304 is merged or closed.

Reply:
fixed 🙏